Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -529,45 +529,87 @@ extension BrowserViewController: WKNavigationDelegate {
     return .allow
   }
 
-  // When this function is moved to Async-Await, do NOT forget to move the `TabManagerNavDelegate` to async-await!
-  public func webView(_ webView: WKWebView, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+  @MainActor
+  public func webView(_ webView: WKWebView, respondTo challenge: URLAuthenticationChallenge) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
 
     // If this is a certificate challenge, see if the certificate has previously been
     // accepted by the user.
     let origin = "\(challenge.protectionSpace.host):\(challenge.protectionSpace.port)"
     if challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust,
       let trust = challenge.protectionSpace.serverTrust,
       let cert = SecTrustGetCertificateAtIndex(trust, 0), profile.certStore.containsCertificate(cert, forOrigin: origin) {
-      completionHandler(.useCredential, URLCredential(trust: trust))
-      return
+      return (.useCredential, URLCredential(trust: trust))
+    }
+    
+    // Certificate Pinning
+    if challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
+      if let serverTrust = challenge.protectionSpace.serverTrust {
+        let host = challenge.protectionSpace.host
+        let port = challenge.protectionSpace.port
+        
+        return await Task.detached(priority: .userInitiated) {
```

Comment:
Why detach? Is `BraveCertificateUtility.verifyTrust` a heavy function?

Reply:
It has to be called on a non-main queue because `verifyTrust` also uses `SecTrustEvaluate` which is required to be called on a non-main queue.